README.md

```diff
@@ -1,106 +1,35 @@
-# The LLVM Compiler Infrastructure
```

Comment:
Last time we moved the build instruction in the mlir-clang directory, and we kept this README. I would prefer to keep this file and move these changes in the mlir-clang dir.